File: file-handling/handling-text-files/python/append_to_text_file.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4
     

def append_to_file():
    """Example of appending to a text file"""
    file_object = open("twinkle.txt", mode="a")
    file_object.write("\n") 
    file_object.write("When the blazing sun is gone\n")
    file_object.write("When he nothing shines upon,\n")
    file_object.write("Then you show your little light,\n")
    file_object.write("Twinkle, twinkle, all the night.\n")
    file_object.close()

    
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    append_to_file()

File: file-handling/handling-text-files/c-sharp/append_to_text_file.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;

namespace IsaacCodeSamples
{
    class Program
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            AppendToFile();
        }


        // Example of appending to a text file
        public static void AppendToFile() {
            using (StreamWriter writer = new StreamWriter("twinkle.txt", true)) {
                writer.WriteLine("");
                writer.WriteLine("When the blazing sun is gone");
                writer.WriteLine("When he nothing shines upon,");
                writer.WriteLine("Then you show your little light,");
                writer.WriteLine("Twinkle, twinkle, all the night.");
            } // The stream is now closed
        }

        
    }
}